src/base/faucet/Faucet.ts

```diff
@@ -0,0 +1,56 @@
+import * as ethers from 'ethers';
+
+import type { Config } from '@app/config';
+import { assert } from '@app/error';
+import type { TransactionResponse } from '@ethersproject/providers';
+import { parseUnits } from '@ethersproject/units';
+
+export async function withdraw(amount: number, config: Config): Promise<TransactionResponse> {
+  assert(config.signer);
+
+  const faucet = new ethers.Contract(
+    config.radToken.faucet,
+    config.abi.faucet,
+    config.signer
+  );
+
+  return faucet.withdraw(config.radToken.address, ethers.utils.parseUnits(amount.toString()));
+}
+
+export async function getMaxWithdrawAmount(config: Config): Promise<ethers.BigNumber> {
+  assert(config.signer);
+
+  const faucet = new ethers.Contract(
+    config.radToken.faucet,
+    config.abi.faucet,
+    config.signer
+  );
+
+  return faucet.maxWithdrawAmount();
+}
+
+export async function calculateTimeLock(amount: ethers.BigNumber, config: Config): Promise<ethers.BigNumber> {
+  assert(config.signer);
+
+  const faucet = new ethers.Contract(
+    config.radToken.faucet,
+    config.abi.faucet,
+    config.signer
+  );
+
+  return faucet.calculateTimeLock(parseUnits(amount.toString()));
```

Comment:
What's this toString/parse about?

Reply:
The whole thing with passing `toString` and then `parseUnits` is because the user will be providing the value in ether RAD while the values obtained and required by the contract are needed in wei RAD.
I think the easiest way is to already parse everything to wei so we have it ready when needed to send to the contract.

Reply:
Ah gotcha. Can you maybe create a helper `toWei` that does this, so it's a bit clearer?

Reply:
Working with BigNumbers, made me appreciate strings for big numbers :smile: 
So I changed everything back to string, and I think it made all much more clearer.